mlflow/utils/autologging_utils/__init__.py

```diff
@@ -591,3 +591,11 @@ def get_method_call_arg_value(arg_index, arg_name, default_value, call_pos_args,
         return call_pos_args[arg_index]
     else:
         return default_value
+
+
+def is_generator(obj):
```

Comment:
Nit: I think this should be named `is_iterator`, since the checks implemented are related to iterator attributes. This *is* the right thing to check, but we should adjust the naming.